tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1605,24 +1604,182 @@ def _test_memory_based_scheduling_enabled_false(
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31129")
 
-    # TODO: Add functional tests for memory-based scheduling
+    # Upload files for memory allocation tests
+    remote_command_executor._copy_additional_files(
+        [
+            str(test_datadir / "memory_allocation_chars.c"),
+        ],
+    )
+
+    # Compile C program to test memory allocations
+    remote_command_executor.run_remote_command("gcc memory_allocation_chars.c")
+    assert_that(remote_command_executor.run_remote_command("ls ./a.out").stdout).contains("a.out")
+
+    # Check that I can use the `--mem` flag to filter compute nodes
+    # Try to allocate on nodes with not enough memory
+    result = slurm_commands.submit_command(
+        nodes=1,
+        command="sleep 1",
+        constraint="ondemand1-i1",
+        other_options="--mem=4000 --test-only",
+        raise_on_error=False,
+    )
+    assert_that(result.stdout).is_equal_to("allocation failure: Requested node configuration is not available")
+
+    # Check that compatible nodes would be selected
+    result = slurm_commands.submit_command(
+        nodes=1,
+        command="sleep 1",
+        other_options="--mem=4000 --test-only",
+        raise_on_error=False,
+    )
+    assert_that(result.stdout).matches(r"^.*Job \d* to start.*$")
+    assert_that(result.stdout).does_not_contain("ondemand1-i1")
+
+    # Check that the `--mem` option only filters compute nodes instead of managing memory required by jobs
+    job_id_1 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "sleep 30",
+            "other_options": "--mem=2000 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    job_id_2 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "sleep 30",
+            "other_options": "--mem=2000 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    time.sleep(jiff)
+    assert_that(slurm_commands.get_job_info(job_id_1, field="JobState")).is_equal_to("RUNNING")
+    assert_that(slurm_commands.get_job_info(job_id_2, field="JobState")).is_equal_to("RUNNING")
+    # Here two jobs submitted with `--mem=2000` can fit on a node with less than 4000 MiB memory
+    # because without memory as consumable resource, Slurm doesn't track the memory usage of
+    # each job.
+    assert_that(slurm_commands.get_job_info(job_id_1, field="NodeList")).is_equal_to(
+        slurm_commands.get_job_info(job_id_1, field="NodeList")
+    )
+    slurm_commands.wait_job_completed(job_id_1)
+    slurm_commands.wait_job_completed(job_id_2)
+
+    # Check that without memory constraining, jobs might contend memory on the compute node
```

Comment:
the command below has `--mem`, so why are you saying "without memory constraining"?
Correct me if I'm wrong, but I'd rather say "without memory as a consumable resource"

Reply:
This is subtle. The thing that actually prevents memory contention is the memory constraining (i.e. `ConstrainRAMSpace=yes`). Though, memory constraining makes sense only if memory is set as a consumable resource. 

So what you say is right, but my comment is more precise. I will make it clearer in the comment.